server/src/main/java/io/spine/server/commandbus/CommandScheduler.java

```diff
@@ -87,12 +85,12 @@ public boolean accept(CommandEnvelope envelope, StreamObserver<Response> respons
         return true;
     }
 
-    private void scheduleAndStore(Command command, StreamObserver<Response> responseObserver) {
+    private void scheduleAndStore(Command command, StreamObserver<Command> ackObserver) {
```

Comment:
I would have a `StreamObserver<MessageAcked>`, where `MessageAcked` is a message:
```

message MessageAcked {

    // An identifier of the message, acknowledged and successfully received by the bus.
    Any message_id = 1 [(required) = true];
}
```
, where in fact `message_id` would be either `CommandId`, `EventId` or `FailureId`, packed as `Any`.

Also, that's just an `observer`.